kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/storage/AwsS3DirectoryLister.scala

```diff
@@ -98,7 +109,11 @@ object AwsS3DirectoryLister extends LazyLogging {
                 if (levelsToRecurse > 0) {
                   acc + prefix
                 } else {
-                  if (connectorTaskId.ownsDir(prefix) && !exclude.contains(prefix)) acc + prefix
+                  if (
+                    connectorTaskId.ownsDir(prefix) && !exclude.contains(prefix) && !wildcardExclude.exists(we =>
+                      prefix.contains(we),
```

Comment:
isn't it enough to filter on level 0 and ignore the other parts? if we exclude the wildcard the code won't end up including keys under `.index/*`.